crates/store/re_protos/proto/rerun/v0/redap.proto

```diff
@@ -0,0 +1,275 @@
+syntax = "proto3";
+
+package rerun.redap.v0;
+
+import "rerun/v0/common.proto";
+
+// The Rerun Data Platform (redap) services are a common set of interfaces for accessing and managing Rerun data.
+//
+// There are 3 key concepts that are referenced in these API definitions, each giving rise to their own subset of
+// service definitions:
+// - Resource: A resource is any data that can be mapped to a single Recording or Table in the Rerun data model.
+//   - For example, a single Rerun recording, an MCAP file, parquet table, etc.
+//   - Every resource also has a set of "Properties" associated with it. This is static data that provides information
+//     about the resource. For example, serial numbers, task descriptions, or other use-defined tags.
+// - Collection: A collection contains multiple resources of a similar type.
+//   - Much of the data that we work with is "episodic" in nature, containing hundreds or thousands of recordings
+//     that are all related to a single task or experiment. There are significant benefits to being able to group
+//     these together as a logical unit when it comes to tasks like search and indexing.
+//   - The easiest way to think of a Collection is as represented by a Table, where each row in the table corresponds
+//     to a resource, and each column corresponds to a shared property of those resources.
+//   - Collections might be provided by "Data Sources" such as a folder of MCAP files, dedicated Rerun storage nodes,
+//     or even views synthesized from other collections in the catalog.
+// - Catalog: The top-level of organization of everything you can access via the Rerun Data Platform.
+//   - The catalog is primarily organized into into a set of named "Entries"
+//   - Entries may be data-sources, collections, views, or other constructs we may add in the future.
+//     - What an entry is, where it gets its data, whether it's indexed, how it exposes its schema, etc. are all
+//     bits of information managed by the catalog that define how the system works
+//
+// The hosted Rerun Data Platform exposes a full implementation of all of these APIs as part of a comprehensive
+// data Catalog experience.
+//
+// However, other Rerun-compatible services may implement a minimal subset of these services for convenience.
+// This makes it possible to directly connect to these services with the Rerun Viewer in minimal
+// deployments without the need to set up a separate catalog server.
+//
+// In such a minimal deployment, the exposed Catalog might consist of a single Collection while the Add/RemoveEntry
+// APIs are unimplemented.
+
+// The Catalog service is for listing and management of multiple catalog entries.
+service Catalog {
+    // List all the available entries in the Catalog
+    //
+    // This must be implemented by all Catalog-compatible services.
+    rpc ListEntries(ListEntriesRequest) returns (ListEntriesResponse) {}
```

Comment:
Maybe consider the `FindEntry` / `FindEntries` pair of end points as discussed during meet-up?